frontend/src/components/MeetingRoom/Toolbar/Toolbar.tsx

```diff
@@ -60,57 +65,82 @@ const Toolbar = ({
   const isReportIssueEnabled = import.meta.env.VITE_ENABLE_REPORT_ISSUE === 'true';
   const isViewingScreenShare = subscriberWrappers.some((subWrapper) => subWrapper.isScreenshare);
   const isScreenSharePresent = isViewingScreenShare || isSharingScreen;
-  const isSmallViewport = useIsSmallViewport();
   const handleLeave = useCallback(() => {
     if (!disconnect) {
       return;
     }
     disconnect();
   }, [disconnect]);
   const [openEmojiGridDesktop, setOpenEmojiGridDesktop] = useState<boolean>(false);
+  const shownButtons = useShownButtons();
+
+  // An array of buttons available for the toolbar. As the toolbar resizes, buttons may be hidden and moved to the
+  // ToolbarOverflowMenu to ensure a responsive layout without compromising usability.
+  const toolbarButtonArray: Array<ReactElement | false> = [
+    <ScreenSharingButton
+      toggleScreenShare={toggleShareScreen}
+      isSharingScreen={isSharingScreen}
+      isViewingScreenShare={isViewingScreenShare}
+      key="ScreenSharingButton"
+    />,
+    <LayoutButton isScreenSharePresent={isScreenSharePresent} key="LayoutButton" />,
+    <EmojiGridButton
+      isEmojiGridOpen={openEmojiGridDesktop}
+      setIsEmojiGridOpen={setOpenEmojiGridDesktop}
+      isParentOpen
+      key="EmojiGridButton"
+    />,
+    <ArchivingButton key="ArchivingButton" />,
+    isReportIssueEnabled && (
+      <ReportIssueButton
+        isOpen={rightPanelActiveTab === 'issues'}
+        handleClick={toggleReportIssue}
+        key="ReportIssueButton"
+      />
+    ),
+    <ParticipantListButton
+      isOpen={rightPanelActiveTab === 'participant-list'}
+      handleClick={toggleParticipantList}
+      participantCount={participantCount}
+      key="ParticipantListButton"
+    />,
+    <ChatButton
+      isOpen={rightPanelActiveTab === 'chat'}
+      handleClick={toggleChat}
+      key="ChatButton"
+    />,
+  ];
 
   return (
-    <div className="absolute bottom-0 left-0 flex h-[80px] w-full flex-col items-center bg-darkGray-100 p-4 md:flex-row md:justify-between">
-      <div className="flex flex-1 justify-start overflow-hidden pr-2">
-        <TimeRoomNameMeetingRoom />
+    <div className="absolute bottom-0 left-0 flex h-[80px] w-full items-center bg-darkGray-100 p-4 md:flex-row md:justify-between">
+      <div className="flex justify-start overflow-hidden">
+        {shownButtons >= toolbarButtonArray.length && <TimeRoomNameMeetingRoom />}
```

Comment:
consider breaking this expression out into a more clear variable like 'isToolbarCollapsed' or similar 🙏